infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -5,1264 +5,254 @@
-	binary.LittleEndian.PutUint64(hashbuf[:], hash64)
-	data2 := []byte{}
-	data2 = append(data2, am.key[:]...)
-	data2 = append(data2, am.GroupKey(srcAddress)...)
-	data2 = append(data2, hashbuf[:]...)
-
-	hash2 := daghash.DoubleHashB(data2)
-	return int(binary.LittleEndian.Uint64(hash2) % NewBucketCount)
+	return result
 }
 
-func (am *AddressManager) triedAddressBucketIndex(netAddress *appmessage.NetAddress) int {
-	// doublesha256(key + group + truncate_to_64bits(doublesha256(key)) % buckets_per_group) % num_buckets
-	data1 := []byte{}
-	data1 = append(data1, am.key[:]...)
-	data1 = append(data1, []byte(NetAddressKey(netAddress))...)
-	hash1 := daghash.DoubleHashB(data1)
-	hash64 := binary.LittleEndian.Uint64(hash1)
-	hash64 %= triedBucketsPerGroup
-	var hashbuf [8]byte
-	binary.LittleEndian.PutUint64(hashbuf[:], hash64)
-	data2 := []byte{}
-	data2 = append(data2, am.key[:]...)
-	data2 = append(data2, am.GroupKey(netAddress)...)
-	data2 = append(data2, hashbuf[:]...)
-
-	hash2 := daghash.DoubleHashB(data2)
-	return int(binary.LittleEndian.Uint64(hash2) % TriedBucketCount)
-}
-
-// addressHandler is the main handler for the address manager. It must be run
-// as a goroutine.
-func (am *AddressManager) addressHandler() {
-	dumpAddressTicker := time.NewTicker(dumpAddressInterval)
-	defer dumpAddressTicker.Stop()
-
-out:
-	for {
-		select {
-		case <-dumpAddressTicker.C:
-			err := am.savePeers()
-			if err != nil {
-				panic(errors.Wrap(err, "error saving peers"))
-			}
-
-		case <-am.quit:
-			break out
-		}
-	}
-	err := am.savePeers()
-	if err != nil {
-		panic(errors.Wrap(err, "error saving peers"))
-	}
-	am.wg.Done()
-	log.Trace("Address handler done")
-}
-
-// savePeers saves all the known addresses to the database so they can be read back
-// in at next run.
-func (am *AddressManager) savePeers() error {
-	serializedPeersState, err := am.serializePeersState()
-	if err != nil {
-		return err
-	}
-
-	return dbaccess.StorePeersState(am.databaseContext, serializedPeersState)
-}
-
-func (am *AddressManager) serializePeersState() ([]byte, error) {
-	peersState, err := am.PeersStateForSerialization()
-	if err != nil {
-		return nil, err
-	}
-
-	buffer := &bytes.Buffer{}
-	encoder := gob.NewEncoder(buffer)
-	err = encoder.Encode(&peersState)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to encode peers state")
-	}
-
-	return buffer.Bytes(), nil
+// New returns a new Kaspa address manager.
+func New(cfg Configer) (*AddressManager, error) {
+	return &AddressManager{
+		addresses: map[AddressKey]*netAddressWrapper{},
+		random:    randomaddress.NewAddressRandomize(),
+		cfg:       cfg,
+	}, nil
 }
 
-// PeersStateForSerialization returns the data model that is used to serialize the peers state to any encoding.
-func (am *AddressManager) PeersStateForSerialization() (*PeersStateForSerialization, error) {
+// AddAddresses adds addresses to the address manager
+func (am *AddressManager) AddAddresses(addresses ...*appmessage.NetAddress) {
```

Comment:
Please add the `AddAddress` function as specified. Then use it inside the loop of AddAddresses.

Note: You will have to have two version of AddAddress: one that locks am.mutex (for the exported AddAddress) and another one that doesn't lock `am.mutex` (for usage inside AddAddresses loop).
In such cases we usually name the non-locking function `addAddressNoLock`.